apps/namegraph.dev/app/name/[name]/name-details-page.tsx

```
@@ -0,0 +1,882 @@
+"use client";
+/* eslint-disable react-hooks/exhaustive-deps */
+
+import {
+  NameGraphCollection,
+  NameGraphFetchTopCollectionMembersResponse,
+  NameGraphSortOrderOptions,
+} from "@namehash/namegraph-sdk/utils";
+import { useEffect, useState } from "react";
+import Skeleton from "@/components/skeleton";
+import { Button } from "@/components/ui/button";
+import { useQueryParams } from "@/components/use-query-params";
+import { ChevronLeft, ChevronRight, Loader } from "lucide-react";
+import { CollectionCard } from "@/components/collections/collection-card";
+import {
+  findCollectionsByMember,
+  findCollectionsByString,
+  FromNameGraphSortOrderToDropdownTextContent,
+  getCollectionsForQuery,
+} from "@/lib/utils";
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { Link } from "@namehash/namekit-react";
+import { NameWithDefaultSuffix } from "@/components/collections/name-with-default-suffix";
+import { CollectionsCardsSkeleton } from "@/components/collections/collections-grid-skeleton";
+import { ens_normalize } from "@adraffy/ens-normalize";
+
+interface NavigationConfig {
+  itemsPerPage: number;
+  totalItems?: Record<NameRelatedCollectionsTabs, number | undefined>;
+}
+
+interface CollectionsData {
+  sort_order: NameGraphSortOrderOptions;
+  related_collections: NameGraphCollection[];
+}
+
+const NameRelatedCollectionsTabs = {
+  ByMembership: "ByMembership",
+  ByConcept: "ByConcept",
+} as const;
+
+type NameRelatedCollectionsTabs =
+  (typeof NameRelatedCollectionsTabs)[keyof typeof NameRelatedCollectionsTabs];
+
+export const NameDetailsPage = ({ name }: { name: string }) => {
+  const normalizedName = ens_normalize(decodeURI(name));
```

If the value returned by `decodeURI` is not equal to the value returned by `ens_normalize` then we should throw an error that we only show name details pages for names that are already normalized.

---

@lightwalker-eth isn't it better UX to show invalid name NFT Avatar and to expose related collections and ideated names that are alike the search done?

---

@FrancoAguzzi I'm open to that. Thank you :+1: